### ISSUE TITLE
Beware of inline scripts

### ISSUE THREAD

**Issue by zymase** [uBlockOrigin/uAssets#468]
uBlockOrigin 1.13.0 / Firefox 52 ESR 52.2.0 (64bit) / Windows 7 (64bit)
uBlockOrigin mode: hard (3rd-party scripts, 3rd-party frames, 3rd-party = globally blocked)

I've found a site that includes a connection via inline scripts: https://www.nextinpact.com/

I've discovered this by accident.
I happen to use an application called PeerBlock to which I've included a list which handles the range:
Microsoft IP Public Ranges 40.64.0.0 - 40.127.255.255

I was surprised to notice that after accessing https://www.nextinpact.com/ PeerBlock showed that the following address had been blocked: 40.118.56.136:443
Normal since it is in the range of the above mentioned filter.

Now, if I block inline scripts for www.nextinpact.com the connection to 40.118.56.136:443 does not appear.
Hence, the connection to 40.118.56.136:443 is performed via the site's inline script.

I checked www.nextinpact.com with unmaskparasites.com:
http://www.unmaskparasites.com/security-report/?page=https://www.nextinpact.com/

Result: 
suspicious inline script found. 
`sas_tmstp=Math.round(Math.random()*10000000000);sas_masterflag=1;function SmartAdServer(sas_pageid,...`

My point: I'm wondering if blocking inline scripts by default is pertinent in the same way blocking (by default) 3rd-party scripts, 3rd-party frames, 3rd-party is.

uBlockOrigin does its job fine, as always.

**Comment by gotitbro** [comment]
Blocking inline scripts would lead to breaking the functionality of many websites as most use it for legitimate purposes.

@gorhill Also, invalid issue.

**Comment by zymase** [comment]
Blocking 3rd-party scripts, 3rd-party frames, 3rd-party leads as well to breaking the functionality of many sites and the very purpose of uBlockOrigin is to consider modes and let the user opt on per-site basis.

I'm not saying -- far from that -- that blocking in-line scripts should be a default setting, only wondering if it should or not be emphasized by including it in the hard mode, or in a whatever hard+ mode. Note that it's always the user's choice, these modes are always only indicative. Whatever seems to me -- given my experience/example above mentioned -- that the inline scripts blocking is worth being emphasized.

**Comment by gotitbro** [comment]
@zymase _Inline scripts_ can already be blocked on a per site basis through Dynamic filtering (https://github.com/gorhill/uBlock/wiki/Dynamic-filtering:-quick-guide). If you want to block inline scripts globally you can use a simple filter like: `*$inline-script`

And if some domain breaks and you want to exclude it simply add it to the above filter like: `,domain=~domain.tld`

**Comment by zymase** [comment]
@gotitbro thanks. I'm aware that inline scripts can already be blocked on a per site basis through Dynamic filtering. That's what I had done in the experience I mentioned above. 

My purpose here was only to express my surprise when discovering that a connection to a 3rd-party site had been established without that connection appearing in uBO's drop-down menu. I mentioned that I had noticed that blocking the concerned site's in-line scripts (per site : for that site, nextinpact.com) blocked the connection.

I then edited my comment to make it clear that the culprit was not the inline script as such but that script given a connection established by tim.nextinpact.com was active, and that blocking tim.nextinpact.com blocked the connection to the 3rd-party site.

The point is that the connection to the 3rd-party site was dependent of both tim.nextinpact.com and the in-line script, that blocking either would block the connection, and that I initially presumed the culprit was the in-line script alone.

Thanks again. Next time I'll further investigate before opening a thread but, as I believed to have made it clear, it was in no way to be understood as an issue. uBlockOrigin has always fulfilled my expectations. I only meant to describe a surprise and found no other way than to express it in this area dedicated to issues, but where else.

All is clear now. Sorry again for this long, far too long litany.
